daemon/graphdriver/overlay/overlay.go

```diff
@@ -263,6 +300,20 @@ func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 		}
 	}()
 
+	driver := &Driver{}
+	if opts != nil {
+		if err := d.parseStorageOpt(opts.StorageOpt, driver); err != nil {
+			return err
+		}
+	}
+
+	if driver.options.quota.Size > 0 && !strings.HasSuffix(dir, "-init") {
+		// enforce container disk project quota
+		if err := d.quotaCtl.SetQuota(dir, driver.options.quota); err != nil {
```

Comment:
@dmcgowan I tried the test and got the same result.
If you set the quota in the upper directory, the quota will not work.
However, setting the quota on the root of the container does not seem to make quota behave differently from overlay2.
The space quoted by quota is only for writing to the upper directory, and lower is not used.
My guess is that the path to which the container is mounted is already the path where the upper and lower are merged by the kernel overlay driver
Please let me know if there is anything I should investigate or test.